cumulusci/robotframework/pageobjects/BasePageObjects.py

```diff
@@ -136,6 +138,42 @@ def populate_form(self, *args, **kwargs):
         # library.
         self.salesforce.populate_form(*args, **kwargs)
 
+    @capture_screenshot_on_error
+    def select_dropdown_value(self, label, value):
+        """Sets the value of a dropdown form field from one of the values in the dropdown
+
+        ``label`` represents the label on the form field (eg: Lead Source)
+
+        If a modal is present, the modal will be searched for the dropdown. Otherwise
+        the first matching dropdown on the entire web page will be used.
+        """
+        input_element = self._get_form_element_for_label(label)
+        if input_element is None:
+            raise Exception(f"No form element found with the label '{label}'")
+
+        # SeleniumLibrary's scroll_element_into_view doesn't seem to work
+        # for elements in a scrollable div on Firefox. Javascript seems to
+        # work though.
+        self.selenium.driver.execute_script(
+            "arguments[0].scrollIntoView()", input_element
+        )
+        trigger = input_element.find_element_by_class_name("uiPopupTrigger")
+        trigger.click()
+
+        try:
+            popup_locator = (
+                "//div[contains(@class, 'uiPopupTarget')][contains(@class, 'visible')]"
+            )
+            popup = self.selenium.get_webelement(popup_locator)
+        except ElementNotFound:
+            raise ElementNotFound("Timed out waiting for the popup menu")
```

Comment:
Is it a dropdown or a popup?

Reply:
I used "popup" in this context since I'm waiting for an element with the class `uiPopupTarget` as the result of clicking on a "uiPopupTrigger".

Reply:
Though, if you're referring to the error message, yeah, that should probably be "dropdown".  I'll fix it.